src/main/java/gov/acwi/wqp/etl/activity/ActivityProcessor.java

```diff
@@ -0,0 +1,163 @@
+
+	private final ConfigurationService configurationService;
+
+	@Autowired
+	public ActivityProcessor(ConfigurationService configurationService) {
+		this.configurationService = configurationService;
+	}
+
+	@Override
+	public Activity process(BiodataActivity bdActivity) throws Exception {
+		Activity activity = new Activity();
+
+		String gear = bdActivity.getEffortGear() == null
+				? bdActivity.getSampleGearUsed()
+				: bdActivity.getEffortGear();
+
+		activity.setDataSourceId(configurationService.getEtlDataSourceId());
+		activity.setDataSource(configurationService.getEtlDataSource());
+
+		activity.setSampleMedia(DEFAULT_SAMPLE_MEDIA);
+		activity.setActivityTypeCode(DEFAULT_ACTIVITY_TYPE_CODE);
+		activity.setAssemblageSampledName(DEFAULT_ASSEMBLAGE_SAMPLED_NAME);
+
+		activity.setStationId(bdActivity.getStationId());
+		activity.setSiteId(bdActivity.getSiteId());
+		activity.setEventDate(bdActivity.getSampleCollectionStartTime().toLocalDate());
+		activity.setActivity(bdActivity.getActivity());
+		activity.setOrganization(bdActivity.getOrganization());
+		activity.setSiteType(bdActivity.getSiteType());
+		activity.setHuc(bdActivity.getHuc());
+		activity.setGovernmentalUnitCode(bdActivity.getGovernmentalUnitCode());
+		activity.setGeom(bdActivity.getGeom());
+		activity.setOrganizationName(bdActivity.getOrganizationName());
+		activity.setActivityId(bdActivity.getActivityId());
+
+		activity.setActivityStartTime(
+				processActivityStartTime(
+						bdActivity.getSampleDataSource(),
+						bdActivity.getSampleCollectionStartTime()));
+
+		activity.setActStartTimeZone(
+				processActivityStartTimeZone(
+						bdActivity.getSampleDataSource(),
+						bdActivity.getSampleTimeDatum()));
+
+		activity.setProjectId(bdActivity.getProjectId());
+		activity.setActivityComment(bdActivity.getActivityComment());
+		activity.setActivityReachLength(bdActivity.getActivityReachLength());
+
+		activity.setActivityReachLengthUnit(
+				processActivityReachLengthUnit(
+						bdActivity.getActivityReachLength()));
+
+		activity.setActivityPassCount(
+				processActivityPassCount(
+						bdActivity.getEffortPass()));
+
+		activity.setSampleCollectMethodId(bdActivity.getSampleCollectMethodId());
+		activity.setSampleCollectMethodCtx(bdActivity.getSampleCollectMethodCtx());
+		activity.setSampleCollectMethodName(bdActivity.getSampleCollectMethodName());
+		activity.setActivitySampleCollectMethodDescription(
+				bdActivity.getActivitySampleCollectMethodDescription());
+
+		activity.setSampleCollectEquipName(
+				processSampleCollectEquipName(gear));
+
+		activity.setActivitySampleCollectEquipmentComments(
+				processActivitySampleCollectEquipmentComments(
+						gear,
+						bdActivity.getDwSampleTypeId(),
+						bdActivity.getEffortSubreach(),
+						bdActivity.getEffortPass()));
+
+		return activity;
+	}
+
+	public String processActivityStartTime(String sampleDataSource, LocalDateTime sampleCollectionStart) {
```

Comment:
This may just be me so take with grain of salt, but I prefer using a name like getActivityStartTime. This makes it clearer that this function is merely transforming the inputs to create the returned value.

Reply:
I agree, `process` seems like a less than ideal verb here.  I initially avoided using `get` as the verb because there is an accessor in the `Activity` class of the same name.  Programatically this doesn't matter since their only similarity is their object reference, but it seemed like it could be a temporary point of confusion.  I'm over it though, going with `get`!

Reply:
Also seems like it should be private. Is there a reason it's a public method?

Reply:
Nope!